include/seqan3/range/concept.hpp

```diff
@@ -61,4 +61,68 @@ SEQAN3_CONCEPT forwarding_range = std::ranges::range<type> && requires (type &&
+ *
+ * \details
+ *
+ * A pseudo random access iterator refines the std::forward_iterator and fulfils in addition all syntactic requirements
+ * of a regular std::random_access_iterator except that iterator category is weaker than
+ * std::random_access_iterator_tag.
+ * These iterators do allow jumping within the associated range or accessing an arbitrary element but cannot guarantee
+ * constant time for these operations. Still, the performance is sub-linear. Typical examples are range adaptors that
+ * store additional information on the original sequence within a tree like data structure. Resolving a specific
+ * position can be then achieved in sub-linear time. However, since certain algorithms that require random access are
+ * built on the assumption that random access is constant, the pseudo random access iterator defines at most
+ * the std::bidirectional_iterator_tag for the iterator category and thus might be picked for a weaker overload of these
+ * algorithms. The user can force to use the random access behavior with non-constant access time using the
+ * seqan3::views::pseudo_random_access adaptor. Note that this does not change the complexity of the used operations.
+ *
+ * ### Concepts and doxygen
+ *
+ * The requirements for this concept are given as related functions and type traits.
+ * Types that model this concept are shown as "implementing this interface".
+ */
+//!\cond
+template <typename iterator_t>
+SEQAN3_CONCEPT pseudo_random_access_iterator =
+    std::forward_iterator<iterator_t> &&
+    !std::is_base_of_v<std::random_access_iterator_tag,
+                       typename std::iterator_traits<iterator_t>::iterator_category> &&
+    std::totally_ordered<iterator_t> &&
+    std::sized_sentinel_for<iterator_t, iterator_t> &&
+    requires (iterator_t i, iterator_t const j, std::iter_difference_t<iterator_t> const n)
+{
+    std::same_as<decltype( i += n ), iterator_t &>;
+    std::same_as<decltype( j +  n ), iterator_t>;
+    std::same_as<decltype( n +  j ), iterator_t>;
+    std::same_as<decltype(   --i  ), iterator_t &>;
+    std::same_as<decltype(   i--  ), iterator_t>;
+    std::same_as<decltype( i -= n ), iterator_t &>;
+    std::same_as<decltype( j -  n ), iterator_t>;
+    std::same_as<decltype(  j[n]  ), std::iter_reference_t<iterator_t>>;
+};
+//!\endcond
+
+/*!\interface seqan3::pseudo_random_access_range <>
+ * \extends   std::ranges::forward_range
+ * \brief     This concept checks if a type models a pseudo random access range.
+ *
+ * \details
+ *
+ * A pseudo random access range is a forward range whose iterator type models seqan3::pseudo_random_access_iterator.
+ *
+ * ### Concepts and doxygen
+ *
+ * The requirements for this concept are given as related functions and type traits.
+ * Types that model this concept are shown as "implementing this interface".
+ */
+//!\cond
+template <typename rng_t>
+SEQAN3_CONCEPT pseudo_random_access_range =
```

Comment:
Some random thought with the naming.

`rng | views::pseudo_random_access` makes the iterator of the range (pseudo) random-accessible 
`pseudo_random_access_range` checks whether the iterator of the range is pseudo random-accessible 
`pseudo_random_access_iterator` checks whether the iterator is pseudo random-accessible

Reply:
I am not sure I could follow what your suggestion is here?
But how about calling the first: `views::enforce_random_access`. I think it more clearly states the intent of the view. So pseudo_random_access_range is defined by means of the pseudo_random_access_iterator and the enforce_random_access requires either a random_access_range or a pseudo_random_access_range. Is that more appropriate?